.github/workflows/label-analysis.yml

```diff
@@ -20,10 +20,37 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-      - name: Install Python dependencies
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+          cache: 'pip'
+          cache-dependency-path: '**/requirements*.txt'
+      - name: Install dependencies
         run: pip install -r tasks/requirements.txt
       - name: Auto assign team label
         run: inv -e github.assign-team-label --pr-id='${{ github.event.pull_request.number }}'
+  release-note-check:
+    if: github.triggering_actor != 'dd-devflow[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
❓ question: Why do we need to get the entire history? It looks like we only have a look at the labels and if the PR contains a releasenote file

Reply:
It's because of `reno`. It looks all commits until the last tag, and I don't know if I can specify this (get history until the last tag) easily

Reply:
I think we can reach something nice using
```
git clone --filter=blob:none --sparse git@github.com:DataDog/datadog-agent.git
git sparse-checkout add releasenotes releasenotes/dca
```
That would do a lazy checkout of only requested files. I'll test this on another PR if you agree.